src/status_im/contexts/keycard/effects.cljs

```diff
@@ -93,8 +93,12 @@
    (keycard/init-card (keycard.utils/wrap-handlers args))))
 
 (rf/reg-fx :effects.keycard/generate-and-load-key
- (fn [args]
-   (keycard/generate-and-load-key (keycard.utils/wrap-handlers args))))
+ (fn [{:keys [mnemonic on-failure] :as args}]
+   (-> mnemonic
+       (native-module/validate-mnemonic)
+       (promesa/then #(keycard/generate-and-load-key
```

Comment:
Wouldn't it be better to add the mnemonic validation directly in the function `generate-and-load-key`? Otherwise, all devs will need to remember to call it before `generate-and-load-key`. Or another way to frame the question @Parveshdhull, do you think there's any situation where we shouldn't validate the mnemonic before calling `generate-and-load-key`?

Same question for the function `create-account-from-mnemonic`, is there any scenario where we don't want to validate the mnemonic? If the answer is no, then it's better to call `native-module/validate-mnemonic` inside `native-module/create-account-from-mnemonic`.

Reply:
hi @ilmotta, Thank you very much for reviewing the PR.

> Wouldn't it be better to add the mnemonic validation directly in the function generate-and-load-key?

This is how I originally implemented validation, please check out https://github.com/status-im/status-mobile/pull/22269#discussion_r1987107512

Reply:
Thanks @Parveshdhull, but the the comment doesn't explain why "keycard ns shouldn't know about native-module".

Well, an alternative if we want to keep the `src/keycard/` namespaces purely isolated from native modules is to create another namespace in `src/status_im/` for abstracting keycard better. I think we need to build the abstractions, because having to call a validation function on every place and remember that for an important validation is not ideal.

This point by me is not a blocker for this PR in any way, just a refactoring suggestion.

Reply:
keycard ns just a "pure" wrapper and app should care about seed validation, we should do it in the places where phrase is entered or generated

Reply:
initially we have this problem because we have screens for keycard and they do some magic with seed phrase, but they should return proper validated seed phrase as string

Reply:
Thanks @flexsurfer. Sounds like the best is to abstract within `src/status_im/` then if we want to keep `src/keycard` isolated.